### ISSUE TITLE
Not working with GNOME 3.30

### ISSUE THREAD

**Issue by letorbi** [EasyScreenCast/EasyScreenCast#207]
## Expected Behavior

* When I click on "start recording", recording is started
* When I click on "options", the options window is show

I general: I expect EasyScreenCast to work with GNOME 3.30 like it did with GNOME 3.28.

## Current Behavior

* When I click on "Start recording", all windows freeze (are not clickable or movable) except for the GNOME Shell itself (so clicks on the top bar still show the expected behaviour).
* When I click on "Options", I see a window with the following content:

```
Error: Requiring Gvc, version none: Typelib file for namespace 'Gvc' (any version) not found

Stack trace:
  @/home/letorbi/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/utilaudio.js:12:7
  @/home/letorbi/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/utilgsp.js:17:7
  @/home/letorbi/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:28:7
  _getExtensionPrefsModule@resource:///org/gnome/shell/extensionPrefs/main.js:76:13
  wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
  _selectExtension@resource:///org/gnome/shell/extensionPrefs/main.js:91:31
  wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
  _onCommandLine@resource:///org/gnome/shell/extensionPrefs/main.js:244:17
  wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
  main@resource:///org/gnome/shell/extensionPrefs/main.js:398:5
  @<main>:1:43
```

## Possible Solution
No real idea, but since it worked flawlessly with GNOME 3.28, I think it's just some kind of depreciation issue. Maybe the missing Gvc typelib is pointing to something

## Steps to Reproduce (for bugs)

1. Install Arch Linux with GNOME 3.30
2. Start GNOME on X.org
3. Activate the extension
4. Click on options (the window with the error should pop up)
5. Click on "Start recording"  (the windows freeze)

## Logs

```
[letorbi@kalliope gnome-shell]$ ls -la /usr/lib/gnome-shell/
total 3092
drwxr-xr-x   2 root root    4096 Sep 19 23:01 .
drwxr-xr-x 217 root root  147456 Sep 23 22:54 ..
-rw-r--r--   1 root root   18504 Sep 14 17:47 Gvc-1.0.typelib
-rwxr-xr-x   1 root root   50880 Sep 14 17:47 libgnome-shell-menu.so
-rwxr-xr-x   1 root root 2402264 Sep 14 17:47 libgnome-shell.so
-rwxr-xr-x   1 root root  124536 Sep 14 17:47 libgvc.so
-rwxr-xr-x   1 root root  321216 Sep 14 17:47 libst-1.0.so
-rw-r--r--   1 root root   31532 Sep 14 17:47 Shell-0.1.typelib
-rw-r--r--   1 root root    6084 Sep 14 17:47 ShellMenu-0.1.typelib
-rw-r--r--   1 root root   39640 Sep 14 17:47 St-1.0.typelib
```

```
[letorbi@kalliope ~]$ journalctl /usr/bin/gnome-shell -f
[...]
Sep 24 09:58:35 kalliope gnome-shell[848]: JS WARNING: [/home/letorbi/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/extension.js 268]: Too many arguments to method Clutter.Actor.add_child: expected 1, got 2
Sep 24 09:58:35 kalliope gnome-shell[848]: JS WARNING: [/home/letorbi/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/extension.js 511]: Too many arguments to method Clutter.Actor.add_child: expected 1, got 2
Sep 24 09:58:58 kalliope gnome-shell[848]: JS WARNING: [/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/dialogs.js 134]: reference to undefined property "indicatorCount"
Sep 24 09:58:58 kalliope org.gnome.Shell.desktop[848]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x1e0002c (EasyScreen)
Sep 24 09:58:58 kalliope gnome-shell[848]: STACK_OP_RAISE_ABOVE: window 0x1e0002c not in stack
Sep 24 09:58:58 kalliope gnome-shell[848]: STACK_OP_RAISE_ABOVE: window 0x1e0002c not in stack
```

## Your Environment

* Where did you download the extension?
 [x] Gnome shell extension website
 [x] From Git (Add commit tag)
 (Tried both)
* Gnome shell version:
3.30
* Operating System and version:
Arch
* Display server: Wayland / X11 / Other
X11

**Comment by letorbi** [comment]
It also happens with all other extensions disabled.

**Comment by aleb** [comment]
I confirm this happens on Archlinux.

**Comment by aleb** [comment]
X.org

**Comment by ickedude** [comment]
Confirmed on Debian Buster
```sh
$ loginctl show-session 2 -p Type
Type=wayland
```

**Comment by TheNittam** [comment]
![" onload="alert(1)](https://user-images.githubusercontent.com/16473567/46580923-054d1e00-ca4e-11e8-8bed-28e8195d9beb.png)

Same Error Here :(

**Comment by TheNittam** [comment]
**Display server** : X11
Kali Linux Rolling 2018.4

<?xml version="1.0"?>
```
**<gnome-version>**
 <platform>**3**</platform>
 <minor>**30**</minor>
 <micro>**1**</micro>
 <distributor>**Debian**</distributor>
 <!--<date></date>-->
**</gnome-version>**
```

**Same Error Here:**
![screenshot from 2018-10-07 16-27-51](https://user-images.githubusercontent.com/16473567/46581023-ed769980-ca4f-11e8-9b1b-3bed602bbca8.png)

**Comment by nichtleiter** [comment]
Same error here!

**Comment by letorbi** [comment]
@tobias47n9e Do you have any plans to fix this problem? I was using EasyScreenCast almost daily and would really like to see it back in GNOME 3.30...

**Comment by simonbcn** [comment]
Same problem on Arch Linux.

**Comment by somethingGoneWrong** [comment]
@letorbi  just add this 4 lines on **utilaudio.js** file at start line 12:
```
... CUT
const GIRepository = imports.gi.GIRepository;
GIRepository.Repository.prepend_search_path("/usr/lib/gnome-shell");
GIRepository.Repository.prepend_library_path("/usr/lib/gnome-shell");
GIRepository.Repository.prepend_search_path("/usr/lib64/gnome-shell");
GIRepository.Repository.prepend_library_path("/usr/lib64/gnome-shell");
const Gvc = imports.gi.Gvc;
...CUT
```
if someone have time create a new PR to fix this. thx :)

**Comment by tobias47n9e** [comment]
I will leave this open until the new version is on the extension page again

**Comment by mickaelistria** [comment]
Thanks for fixing it!

> I will leave this open until the new version is on the extension page again

Any ETA about it?

**Comment by chrn0s** [comment]
reinstall easy screen cast , it worked for me

> git clone https://github.com/EasyScreenCast/EasyScreenCast.git
> cd EasyScreenCast
> make
> make install

**Comment by tobias47n9e** [comment]
Seems like the new version got published

**Comment by letorbi** [comment]
Yeah, the issue reported here is fixed, but now #211 is a blocker...
